docs/test_development.rst

```diff
@@ -82,6 +82,61 @@ Moreover the unit tests are executed for `each new pull request via Travis
 CI`_.
 
 
+Functional Tests
+====================
+
+Functional API tests are stored in logical chunks in files. Tests are tagged by tags.
+So it is possible to run for example just Ceph tests. All tests should have
+fixture for proper ``setup`` and ``teardown``. All objects created during testing
+should be removed after test run. There should not be any remains after test run.
+
+There are *only* 2 exceptions:
+* CRUD *happy path* tests which are stored in one file where they share
+  object created and deleted during testing tests from file. These tests should run
+  in same order like they are written in the file. CRUD tests **are not** tagged.
+
+* create or import cluster tests which run at the same beginning of testing because
+  they left created/imported clusters for further testing. This exception exists
+  because cluster creating and importing have extremly big resource needs.
+
+Files with tests example:
+
+* crud_ceph_pool.py - contents create/read/update/delete Ceph pool tests.
+  Tests should run in exact order. Tests share ``pool`` object which is
+  created in 1st test, read in 2nd test, update in 3rd test and deleted
+  in last test. This kind of tests are usually run in CI 
+  for smoke/reggression/acceptance testing. Sharing of ``pool`` object
+  saves resources. These tests are only *positive* or *happy path*.
+  All mentioned tests for one object are stored in one file
+  with prefix crud + [storage type] + tested object.
+
+* create_ceph_cluster.py - In file with name in form (create|import) + [storage type]
+  are stored *positive* or *happy path* tests for importing or creating cluster
+  for one storage type. These tests are run before any other tests because
+  they create or import cluster which is used during further testing.
+  Cluster identification is stored in configuration so it can be used in other tests.
+  For example Ceph cluster is identified by its name and Gluster cluster is identified
```

Comment:
Both gluster and ceph should be identified by one of their machines. Because ceph name doesn't have to be unique. Moreover the name is not displayed during import. Such identifier is needed only for creating/importing of the cluster.
For any reuse of the cluster, we should save Tendrl cluster id. However unfortunately the id is not enough for UI for ceph clusters. For that we have to save also the ceph cluster name.